qubes/storage/file.py

```diff
@@ -358,13 +381,30 @@ def start(self):
             if hasattr(self, 'path_source_cow'):
                 if not os.path.exists(self.path_source_cow):
                     create_sparse_file(self.path_source_cow, self.size)
+
+        async with _lock:
+            if self.save_on_stop:
+                assert not self.snap_on_start, 'unsupported configuration'
+                await qubes.utils.run_program('sudo', CREATE_SCRIPT, self.path,
```

Comment:
Better add sudo only when needed (otherwise it will pollute logs...) - maybe add it inside run_program function?